public/app/features/alerting/unified/components/receivers/grafanaAppReceivers/useReceiversMetadata.ts

```diff
@@ -39,7 +39,7 @@ export function getOnCallMetadata(
   }
 
   const matchingOnCallIntegration = onCallIntegrations.find(
-    (integration) => integration.integration_url === receiver.settings.url
+    (integration) => integration.integration_url === receiver.settings?.url
```

Comment:
Settings field seems to be required according to the existing types. Does it mean it should be marked as optional?
I guess there are more places where we assume that `settings` is required
```js
export type GrafanaManagedReceiverConfig = {
  uid?: string;
  disableResolveMessage: boolean;
  secureFields?: Record<string, boolean>;
  secureSettings?: Record<string, any>;
  settings: Record<string, any>;
```

Reply:
theoretically that was not optional...but given the api is not returning this field..maybe you're right, we should change it

Reply:
Same as this comment? :) https://github.com/grafana/grafana/pull/82132#discussion_r1482978361

Reply:
Setting the property to `optional` gives us only a few places where we access `settings` when in fact it could be `undefined`.

we should definitely either split up the types here or make `settings` optional so the type checker can catch these before we do.